test/common/upstream/least_request_load_balancer_fuzz_test.cc

```diff
@@ -0,0 +1,83 @@
+#include <memory>
+
+#include "test/common/upstream/least_request_load_balancer_fuzz.pb.validate.h"
+#include "test/common/upstream/zone_aware_load_balancer_fuzz_base.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+// Least Request takes into account both weights (handled in ZoneAwareLoadBalancerFuzzBase), and
+// requests active as well
+void setRequestsActiveForStaticHosts(NiceMock<MockPrioritySet>& priority_set,
+                                     const std::string& random_bytestring) {
+  uint32_t index_of_random_bytestring = 0;
+  // Iterate through all the current host sets and set requests for each
+  for (uint32_t priority_level = 0; priority_level < priority_set.hostSetsPerPriority().size();
+       ++priority_level) {
+    MockHostSet& host_set = *priority_set.getMockHostSet(priority_level);
+    for (auto& host : host_set.hosts_) {
+      // Make sure no weights persisted from previous fuzz iterations
+      ASSERT(host->stats().rq_active_.value() == 0);
+      host->stats().rq_active_.set(
+          random_bytestring[index_of_random_bytestring % random_bytestring.length()] % 3);
+      ++index_of_random_bytestring;
+    }
+  }
+}
+
+void removeRequestsActiveForStaticHosts(NiceMock<MockPrioritySet>& priority_set) {
```

Comment:
this feels like a special case of the code above, right?
`"0"`.

Reply:
Oh beautiful, modularity :).

Reply:
I messed around with this, I liked your idea to just call into that, but I think the ASSERT on Host Weight for each fuzz iteration is too valuable to remove. I'll leave as is.